test/models/preprocessor_primo_test.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+#
+require 'test_helper'
+
+class PreprocessorPrimoTest < ActiveSupport::TestCase
+  test 'to_tacos returns unhandled for complex queries' do
+    input = 'any,contains,space;;any,contains,madness'
+
+    assert_equal('unhandled complex primo query', PreprocessorPrimo.to_tacos(input))
+  end
+
+  test 'to_tacos returns unhandled for targeted field queries' do
+    input = 'title,contains,space'
+
+    assert_equal('unhandled complex primo query', PreprocessorPrimo.to_tacos(input))
+  end
+
+  test 'to_tacos returns phrase ready for tacos for simple keyword input' do
+    input = 'any,contains,space'
+
+    assert_equal('space', PreprocessorPrimo.to_tacos(input))
+  end
+
+  test 'to_tacos returns phrase ready for complex keyword input' do
+    input = 'any,contains,Yan, F., Krantz, P., Sung, Y., Kjaergaard, M., Campbell, D.L., Orlando, T.P., Gustavsson, S. and Oliver, W.D., 2018. Tunable coupling scheme for implementing high-fidelity two-qubit gates. Physical Review Applied, 10(5), p.054062.'
+    expected = 'Yan, F., Krantz, P., Sung, Y., Kjaergaard, M., Campbell, D.L., Orlando, T.P., Gustavsson, S. and Oliver, W.D., 2018. Tunable coupling scheme for implementing high-fidelity two-qubit gates. Physical Review Applied, 10(5), p.054062.'
+
+    assert_equal(expected, PreprocessorPrimo.to_tacos(input))
+  end
+
+  test 'keyword? returns true for any contains phrase pattern' do
+    input = 'any,contains,popcorn anomoly'.split(',')
+
+    assert(PreprocessorPrimo.keyword?(input))
+  end
+
+  test 'keyword? returns false for input with more than 3 array elements' do
+    # NOTE: this query entering tacos would work... but it would have been cleaned up prior to running
+    # keyword? in our application via the normal flow
+    input = 'any,contains,popcorn anomoly: why life on the moon is complex, and other cat facts'.split(',')
```

Comment:
See above comment about the calling of this method never being able to match the input of this test - we're explicitly composing the argument as a three-element array, if I follow the logic above?

Reply:
We're composing it in one place that way...but if the method is reused elsewhere or there is a refactor that changes how we are calling it,  it's important to have tests about our expectations for the method independently (unless it's a private method) to ensure we don't break something by changing this behavior.

I'll take a look at this to see if there is a way to make it seem less weird though. I did a sort of late refactor and it's possibly I only half way completed what was in my brain and thus caused this (if you look at the first commit alone and the changes in the second you can probably see how it may have made sense to do it this way at first but might not after the change... I'll check).

Reply:
Good point, and that's a fair reason to have a test like this.